python/ray/data/_internal/block_batching/iter_batches.py

```diff
@@ -97,8 +98,13 @@ def iter_batches(
             process. If set to greater than 0, a separate thread will be used to fetch
             the specified amount of formatted batches from blocks. This improves
             performance for non-CPU bound UDFs, allowing batch fetching compute and
-            formatting to be overlapped with the UDF. Defaults to 0 (no prefetching
-            enabled).
+            formatting to be overlapped with the UDF. Defaults to 1.
+        gpu_prefetch_batches: The number of batches to fetch ahead of the current
+            batch to fetch on the GPU. If set to greater than 0, a separate
+            threadpool will be used to format batches and apply the collate_fn.
+            Defaults to 1. You can revert back to the old prefetching behavior
+            that uses `prefetch_blocks` by setting `use_legacy_iter_batches` to
+            True in the DataContext.
```

Comment:
Given that it's unlikely someone would want this to be >1, I don't think the comment needs to mention the legacy behavior.